### PR TITLE
Simplify DynQuantity

### PR DESCRIPTION
**Pull request by dmcclean**
Simplify the operations that lift into `DynQuantity` by capitalizing on the `Monad` instance for `Maybe`.
Counter-proposal to #117.

### CODE REVIEW

File: src/Numeric/Units/Dimensional/Dynamic.hs [hunk 1]
```diff
@@ -130,7 +129,7 @@ instance Fractional a => Fractional (DynQuantity a) where
   fromRational = demoteQuantity . (*~ one) . P.fromRational
 
 instance Floating a => Floating (DynQuantity a) where
-  pi = DynQuantity . Just $ AnyQuantity D.dOne P.pi
+  pi = demoteQuantity $ P.pi *~ one
```
Comment (bjornbm) [inline src/Numeric/Units/Dimensional/Dynamic.hs]: `demoteQuantity pi`?

**Comment by bjornbm** [comment]
This is more clever, but IMO more difficult to read/understand. And building on the `Maybe` monad won't benefit users since the internals are not exported. I narcissistically prefer my version as being more direct and to the point. :)

File: src/Numeric/Units/Dimensional/Dynamic.hs (hunk 1 above)
Comment (dmcclean) [inline src/Numeric/Units/Dimensional/Dynamic.hs]: Hahaha yup! On your branch too. Whooops!

**Comment by dmcclean** [comment]
That's probably fair. At least it was a good exercise in thinking this way around, which is legitimately useful in other scenarios. You could now or later export the newtype constructor if there was a reasonable scenario for people needing to extend it, but I can't think of any off hand.

We should add an `atan2` implementation though, the question is whether we should do the entire `RealFloat` class (breaking dimensional abstraction) or not.

**Pull request merged**